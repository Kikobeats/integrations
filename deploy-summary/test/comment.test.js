const { createComment } = require('../lib/comment')

const createOpts = (nScreenshots, nOthers, nDeleted) => {
  return {
    commitSha: '<commit_sha>',
    url: '<url>',
    screenshots: Array(nScreenshots)
      .fill()
      .map((_, i) => ({
        route: `<screenshot${i}_route>`,
        routeLink: `<screenshot${i}_route_link>`,
        thumbnailUrl: `<screenshot${i}_thumbnail_url>`,
        screenshotUrl: `<screenshot${i}_screenshot_url>`
      })),
    others: Array(nOthers)
      .fill()
      .map((_, i) => ({
        route: `<other${i}_route>`,
        routeLink: `<other${i}_route_link>`
      })),
    deleted: Array(nDeleted)
      .fill()
      .map((_, i) => `<deleted${i}_route>`)
  }
}

test.each([
  ['simple opts', createOpts(2, 2, 2)],
  ['only others', createOpts(0, 2, 0)],
  ['only screenshots', createOpts(2, 0, 0)],
  ['only deleted', createOpts(0, 0, 2)]
])('%s', (_, opts) => {
  expect(createComment(opts)).toMatchSnapshot()
})
