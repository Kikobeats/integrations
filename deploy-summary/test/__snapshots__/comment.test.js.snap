// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`only deleted 1`] = `
"#### 📝Changed routes:




2 deleted routes:
- **<deleted0_route>**
- **<deleted1_route>**

Commit <commit_sha> (<url>)."
`;

exports[`only others 1`] = `
"#### 📝Changed routes:



- [**<other0_route>**](<other0_route_link>)
- [**<other1_route>**](<other1_route_link>)



Commit <commit_sha> (<url>)."
`;

exports[`only screenshots 1`] = `
"#### 📝Changed routes:


| [**<screenshot0_route>**](<screenshot0_route_link>) | [**<screenshot1_route>**](<screenshot1_route_link>) |
|:-:|:-:|
|<a href=\\"<screenshot0_route_link>\\"><img src=\\"<screenshot0_thumbnail_url>\\" alt=\\"Screenshot of <screenshot0_route>\\" width=\\"300\\"></a><br /><sup><a href=\\"<screenshot0_screenshot_url>\\">(view full size)</a> |<a href=\\"<screenshot1_route_link>\\"><img src=\\"<screenshot1_thumbnail_url>\\" alt=\\"Screenshot of <screenshot1_route>\\" width=\\"300\\"></a><br /><sup><a href=\\"<screenshot1_screenshot_url>\\">(view full size)</a> |







Commit <commit_sha> (<url>)."
`;

exports[`simple opts 1`] = `
"#### 📝Changed routes:


| [**<screenshot0_route>**](<screenshot0_route_link>) | [**<screenshot1_route>**](<screenshot1_route_link>) |
|:-:|:-:|
|<a href=\\"<screenshot0_route_link>\\"><img src=\\"<screenshot0_thumbnail_url>\\" alt=\\"Screenshot of <screenshot0_route>\\" width=\\"300\\"></a><br /><sup><a href=\\"<screenshot0_screenshot_url>\\">(view full size)</a> |<a href=\\"<screenshot1_route_link>\\"><img src=\\"<screenshot1_thumbnail_url>\\" alt=\\"Screenshot of <screenshot1_route>\\" width=\\"300\\"></a><br /><sup><a href=\\"<screenshot1_screenshot_url>\\">(view full size)</a> |



And 2 other routes:
- [**<other0_route>**](<other0_route_link>)
- [**<other1_route>**](<other1_route_link>)

2 deleted routes:
- **<deleted0_route>**
- **<deleted1_route>**

Commit <commit_sha> (<url>)."
`;
